pkg/v1/sdk/features/runner/runner.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+)
+
+// FeatureGateRunner defines methods to interact with FeatureGate resources
+type FeatureGateRunner struct {
+	featureGateClient client.Client
+}
+
+// FeatureInfo is a struct that holds Feature information
+type FeatureInfo struct {
+	Name        string
+	Maturity    string
+	Description string
+	Activated   bool
+	Available   bool
+	Immutable   bool
+}
+
+// NewFeatureGateRunner returns an instance of featureGateRunner.
+func NewFeatureGateRunner() (*FeatureGateRunner, error) {
+	featureGateClient, err := getFeatureGateClient()
+	if err != nil {
+		return nil, err
+	}
+	return &FeatureGateRunner{featureGateClient: featureGateClient}, nil
+}
+
+// getFeatureGateClient returns a new FeatureGate client
+func getFeatureGateClient() (client.Client, error) {
+	var restConfig *rest.Config
+	var err error
+
+	scheme := runtime.NewScheme()
+	if err := configv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	if restConfig, err = k8sconfig.GetConfig(); err != nil {
+		return nil, err
+	}
+	crClient, err := client.New(restConfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create cluster client: %w", err)
+	}
+	return crClient, nil
+}
+
+// GetFeatureGate fetches the FeatureGate
+func (f *FeatureGateRunner) GetFeatureGate(ctx context.Context, featureGateName string) (*configv1alpha1.FeatureGate, error) {
```

Comment:
i prefer to keep the runner package, it would easier for someone to understand that this package contains methods to interact with the featuregate resources by just looking at the name.. moreover moving them to util package gives an impression that they are some helper functions

Reply:
It's fine as a set of utility functions, it's just a client. But if you have strong opinions on keeping the package, I have a few suggestions in other comments.